app/models/miq_provision_virt_workflow.rb

```diff
@@ -395,16 +395,17 @@ def allowed_cat_entries(options)
   end
 
   def allowed_vlans(options = {})
-    @allowed_vlan_cache ||= available_vlans(options)
+    @allowed_vlan_cache ||= available_vlans_and_hosts(options)[0]
     filter_by_tags(@allowed_vlan_cache, options)
   end
 
-  def available_vlans(options = {})
+  def available_vlans_and_hosts(options = {})
     @vlan_options ||= options
     vlans = {}
     src = get_source_and_targets
     return vlans if src.blank?
 
+    hosts = []
```

Comment:
@lfu With the current code you want `hosts` to be nil so `allowed_dvs` calls `get_selected_hosts`.  But I think we can remove a little duplication if we change the code so `available_vlans_and_hosts` always loads the hosts (regardless of the `:vlans` flag and pass that back for use in `allowed_dvs`.

It is unlikely that vlan collection would be turned off here, the real usage is disabling dvs, but the loading of hosts would still need to be moved outside of the `@vlan_options[:vlans]` check.

Thoughts?